specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-11-15/restorable.json

```diff
@@ -0,0 +1,1815 @@
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The unique resource Identifier of the ARM resource."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the ARM resource."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The type of Azure resource."
+        }
+      }
+    },
+    "RestorableMongodbCollectionProperties": {
+      "description": "The properties of an Azure Cosmos DB MongoDB collection event",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "description": "The resource of an Azure Cosmos DB MongoDB collection event",
+          "type": "object",
+          "properties": {
+            "_rid": {
+              "type": "string",
+              "readOnly": true,
+              "description": "A system generated property. A unique identifier."
+            },
+            "operationType": {
+              "type": "string",
+              "readOnly": true,
+              "description": "The operation type of this collection event.",
+              "$ref": "#/definitions/OperationType"
+            },
+            "eventTimestamp": {
+              "type": "string",
+              "readOnly": true,
+              "description": "The time when this collection event happened."
+            },
+            "ownerId": {
+              "type": "string",
+              "readOnly": true,
+              "description": "The name of this MongoDB collection."
+            },
+            "ownerResourceId": {
+              "type": "string",
+              "readOnly": true,
+              "description": "The resource ID of this MongoDB collection."
+            }
+          }
+        }
+      }
+    },
+    "RestorableMongodbResourcesListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "../../stable/2021-11-15/cosmos-db.json#/definitions/DatabaseRestoreResource"
+          },
+          "description": "List of restorable MongoDB resources, including the database and collection names."
+        }
+      },
+      "description": "The List operation response, that contains the restorable MongoDB resources."
+    },
+    "RestorableGremlinDatabasesListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RestorableGremlinDatabaseGetResult"
+          },
+          "description": "List of Gremlin database events and their properties."
+        }
+      },
+      "description": "The List operation response, that contains the Gremlin database events and their properties."
+    },
+    "RestorableGremlinDatabaseGetResult": {
```

Comment:
>RestorableGremlinDatabaseGetResult [](http://example.com/codeflow?start=5&length=34)

Please set x-ms-azure-resource to true. Also consider referencing the definition in /specification/common-types/resource-management/v3/types.json

Reply:
Hi Jin,
We have similar API already existing for Sql/MongoDB (https://github.com/Azure/azure-rest-api-specs/blob/569bb5b5926008180266346d4af44647363446d4/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-11-15/restorable.json#L305).

We are just adding support for Tables and Gremlin for same restorableResources Api.